addons/report/models/report.py

```diff
@@ -76,6 +76,147 @@ def _get_wkhtmltopdf_bin():
+
+    if paperformat.page_height and paperformat.page_width and paperformat.format == 'custom':
+        command_args.extend(['--page-width', str(paperformat.page_width) + 'mm'])
+        command_args.extend(['--page-height', str(paperformat.page_height) + 'mm'])
+
+    if specific_paperformat_args and specific_paperformat_args.get('data-report-margin-top'):
+        command_args.extend(['--margin-top', str(specific_paperformat_args['data-report-margin-top'])])
+    else:
+        command_args.extend(['--margin-top', str(paperformat.margin_top)])
+
+    if specific_paperformat_args and specific_paperformat_args.get('data-report-dpi'):
+        command_args.extend(['--dpi', str(specific_paperformat_args['data-report-dpi'])])
+    elif paperformat.dpi:
+        if os.name == 'nt' and int(paperformat.dpi) <= 95:
+            _logger.info("Generating PDF on Windows platform require DPI >= 96. Using 96 instead.")
+            command_args.extend(['--dpi', '96'])
+        else:
+            command_args.extend(['--dpi', str(paperformat.dpi)])
+
+    if specific_paperformat_args and specific_paperformat_args.get('data-report-header-spacing'):
+        command_args.extend(['--header-spacing', str(specific_paperformat_args['data-report-header-spacing'])])
+    elif paperformat.header_spacing:
+        command_args.extend(['--header-spacing', str(paperformat.header_spacing)])
+
+    command_args.extend(['--margin-left', str(paperformat.margin_left)])
+    command_args.extend(['--margin-bottom', str(paperformat.margin_bottom)])
+    command_args.extend(['--margin-right', str(paperformat.margin_right)])
+    if paperformat.orientation:
+        command_args.extend(['--orientation', str(paperformat.orientation)])
+    if paperformat.header_line:
+        command_args.extend(['--header-line'])
+
+    return command_args
+
+
+def run_wkhtmltopdf(headers, footers, bodies, landscape, paperformat, spec_paperformat_args=None, set_viewport_size=False):
+    """Execute wkhtmltopdf as a subprocess in order to convert html given in input into a pdf
+    document.
+
+    :param header: list of string containing the headers
+    :param footer: list of string containing the footers
+    :param bodies: list of string containing the reports
+    :param landscape: boolean to force the pdf to be rendered under a landscape format
+    :param paperformat: ir.actions.report.paperformat to generate the wkhtmltopf arguments
+    :param specific_paperformat_args: dict of prioritized paperformat arguments
+    :returns: paths to documents_paths
+    """
+
+    command_args = []
+    if set_viewport_size:
+        command_args.extend(['--viewport-size', landscape and '1024x1280' or '1280x1024'])
+
+    # Passing the cookie to wkhtmltopdf in order to resolve internal links.
+    try:
+        if request:
+            command_args.extend(['--cookie', 'session_id', request.session.sid])
+    except AttributeError:
+        pass
```

Comment:
Try to not use pass after an except, use _logger.info It is better and more informative.